### ISSUE TITLE
RCS1197 does not recognize Constant Interpolated Strings

### ISSUE THREAD

**Issue by jnyrup** [JosefPihrt/Roslynator#898]
**Product and Version Used**: 
V4.1.0

**Steps to Reproduce**:
C# 10 introduced [Constant Interpolated Strings
](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/constant_interpolated_strings) such that the snippet below using `Append(string)` does not use runtime string interpolation.
```cs
const string Foo = "Foo";
new StringBuilder().Append($"{Foo}Bar");
```

[SharpLab example](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAcoKgW1QOzgBsiBjACwlIGsBLfAcwB8ABABgAIWBGAOgBUIADxgBuALAAoFgGYuAJg4BhKQG8pHTVzksALBwDKAexwQAshBjkjAEwAUASg1b1kre46kj+ZDC7dOADEjIw4AXg4AImCjSIk3D018CAB3QxgEegYAIThaIhsIBEdeAEF0dAh8ewASSNUYgF9srEiHePdGqUagA===)

![image](https://user-images.githubusercontent.com/78789299/165909646-f4b86369-7a3f-46f7-85c0-7ed0d64d5d03.png)
![image](https://user-images.githubusercontent.com/78789299/165909592-75b50bbc-6602-4d00-a38c-6530ea457c02.png)

This looks related to #658 which was about non-interpolated string constants.

**Actual Behavior**:
RCS1197 is reported.

**Expected Behavior**:
RCS1197 should not be reported.

**Comment by jnyrup** [comment]
Thanks!

**Comment by JosefPihrt** [comment]
NP

**Comment by fgimian** [comment]
@JosefPihrt Sorry to comment on a closed issue, I'm using Roslynator.Analyzers 4.2.0 and still seem to be getting this warning in the same situation:

![image](https://user-images.githubusercontent.com/1811813/222871191-a270ff2f-454c-43a9-95b6-d6faea0a6ff4.png)

Any ideas what I could be missing?

Thanks a lot
Fotis

**Comment by jnyrup** [comment]
@fgimian
Which framework are you targeting - to know what overloads of `StringBuilder.Append` are available.
Which overload of `StringBuilder.Append` does the code resolve to?
What is `shellFontTitle`?
If it's not a constant, such that `$"{shellFontTitle} (TrueType)"` is a Constant Interpolated String, then your issue sounds like #899. (I case you forgot that you already commented on that issue 🙃).

**Comment by fgimian** [comment]
> @fgimian Which framework are you targeting - to know what overloads of `StringBuilder.Append` are available. Which overload of `StringBuilder.Append` does the code resolve to? What is `shellFontTitle`? If it's not a constant, such that `$"{shellFontTitle} (TrueType)"` is a Constant Interpolated String, then your issue sounds like #899. (I case you forgot that you already commented on that issue 🙃).

Oh so sorry, I totally forgot about that other issue, it's been a while since I looked at this codebase 😄 

To answer your question, I'm on .NET 6.0.406.  The overload it's resolving to is:

![image](https://user-images.githubusercontent.com/1811813/222896556-0f2b837e-966a-448a-ab10-6c8682acbf5b.png)

The variable `shellFontTitle` is being computed as follows:

```c#
        string? shellFontTitle = null;
        try
        {
            ShellFile shellFile = new(path);
            shellFontTitle = shellFile.Properties.System.Title.Value;
        }
        catch (FileNotFoundException)
        {
            // Inexistent files will be handled below by checking whether the font title is set.
        }
```

Sounds like my issue is related to #899, please let me know if I'm mistaken 😄 

Thanks again!!
Fotis
